### ISSUE TITLE
ractive.update() does not update data functions

### ISSUE THREAD

**Issue by kurdin** [ractivejs/ractive#1671]
I have code where I want to update whole UI if `lang` set to new value, instead if `custom_fn` I use `translate` function with return from `I18n-js` library but I removed it from example to make it simple. 

`App.update();` does not trigger `data.custom_fn()` so UI not updated. 

``` javascript
App = new Ractive({
    el: '#app',
    template: '#template',
    data: {
      lang: 'en',
      custom_fn: function(s) {
          return s + ' Sergey current Lang is ' + this.data.lang;
      }
    }
});

App.observe('lang', function(lang) {
    console.log('Current lang is: ' + lang);
    App.update(); 
});

App.set('lang','ru');
```

Template:

``` html
{{ custom_fn('Hello') }}
<p>Lang is: {{ lang }}</p>
```

output is:

```
Hello Sergey current Lang is en
Lang is: ru
```

and shoud be

```
Hello Sergey current Lang is ru
Lang is: ru
```

http://jsfiddle.net/du7v1q1c/

Am I doing something wrong?

**Comment by martypdx** [comment]
If you use the api in the custom function, it works automatically without the observe (see http://jsfiddle.net/du7v1q1c/1/):

```js
App = new Ractive({
    el: '#app',
    template: '#template',
    data: {
      lang: 'en',
      custom_fn: function(s) {
          return s + ' Sergey current Lang is ' + this.get('lang');
      }
    }
});

App.set('lang','ru');
```

Does that work for you?

**Comment by kurdin** [comment]
Thanks @martypdx, this works but in theory that should be working with observe and App.update() as well right? 

Let say I want to change language in ``I18n-js`` only once  when ``lang`` set to another value:

```javascript 
App.observe('lang', function(lang) {
     I18n.locale = lang;
     console.log('Current lang is: ' + I18n.currentLocale());
     App.update();
});
````
with code you provided I need to set ``I18n.locale`` for each template value
````javascript 
model = {
      lang: 'en',
      translate: function(s) {
        I18n.locale = this.get('lang');
        return I18n.t(s);
      }
};
````
This will work but not efficient as with observe, where I change ``I18n.locale`` only once and then update UI with ``App.update();``

**Comment by martypdx** [comment]
Ah, I see. If you use the translator directly (or place the function off an object), you can get it to work by using the object as the update keypath (see http://jsfiddle.net/du7v1q1c/3/):

```html
{{ i18n.t('Hello') }}
<p>Lang is: {{ lang }}</p>
```
```js
App = new Ractive({
    el: '#app',
    template: '#template',
    data: {
      lang: 'en',
      i18n: Translator
    }
});

App.observe('lang', function(lang) {
    Translator.local = lang;
    App.update('i18n'); 
});

App.set('lang','ru');
```

I'm marking as bug because computed properties can be explicitly updated, so it follows that data functions should work that way as well. 

BTW, if you're worried about efficiency, using `App.update()` without a keypath checks __all__ keypaths, so even if function updates worked, you would want to specifically use `App.update('translate')` to target the update only to the function.

**Comment by kurdin** [comment]
@martypdx this is absolutely fantastic! :clap:   

Gentlemen, if you need full example how to do localization with Ractive based on https://github.com/fnando/i18n-js check this out:
http://jsfiddle.net/ejv3xe0e/ 

>BTW, if you're worried about efficiency, using App.update() without a keypath checks all keypaths

yes I know, ``App.update('translate')`` :+1:

**Comment by Madgvox** [comment]
I prematurely closed this. Even though @kurdin's problem was solved, as of 0.8-edge data functions still cannot be force-updated using `ractive.update`

**Comment by maperokov** [comment]
Hi all, I think I have the similar problem. You can find the code <a href="http://jsfiddle.net/gLyrugmr/2/">here</a>.

As you can see, I have a helper function which returns one value (in this example, the 'true'), but when a change occure from another module (that's simulated here with this 'counter' variable), it should return another value.

So, what I want to do is to update the part of DOM without the actual model change, the only thing that is different is the output of the helper method.

Is this even a good way of doing something like this?
